### ISSUE TITLE
Green Slimes cause red wall of text... seemingly harmless?

### ISSUE THREAD

**Issue by RandomLonelyFox** [daleao/modular-overhaul#65]
Every time an area that is inhabited by monsters (seemingly specifically the green slime) is entered, SMAPI produces a red wall of text that I have honestly no idea on how to decipher. There do not appear to be any actual gameplay issues as I have only ever realised this happen when switching to SMAPI and seeing the error messages. I have successfully recreated this issue multiple times. 

Here is the log for its' latest occurrence: 
https://smapi.io/log/f5fc2be91eea4daab6f33f77aee4ea06

**Comment by daleao** [comment]
This part of the code deals with Piped Slimes; i.e., when you use the Piper Limit Break.
Although I don't see anything in the code that could throw this error.
Can you please share you save folder?

**Comment by RandomLonelyFox** [comment]
Of course, I've uploaded it to Dropbox, Here's the link: 
https://www.dropbox.com/scl/fo/haz2bpejaeq87nmooxnix/h?rlkey=d0omio6iedf1fmbmcoj7ss2oc&dl=0

Thank you for the quick response!

**Comment by daleao** [comment]
Looks like the link is not public. Sent access request.

**Comment by RandomLonelyFox** [comment]
Have done so, I apologise for the issue, this is my first time using dropbox/sharing files online

**Comment by daleao** [comment]
Strange, you're not even have a level 10 combat profession, which means you are definitely not using the Piper Limit Break.
I tried warping into random mine levels and kill some Slimes, and I saw no errors.
On Day 20th, the following mine floors are infested by Slimes:  24, 57, 86.
I slept 2 days until the 2th, then warped to those floors and cleared all the Slimes successfully with no errors.

Please review step-by-step instructions to reproduce your issue. Use the above above information about infested floors on day 20. Or, alternatively, upload your save [here](https://mouseypounds.github.io/stardew-predictor/) to predict other Slime floors. (they are the ones indicated by the Slime icon).
![image](https://github.com/daleao/modular-overhaul/assets/111105627/a0732448-d4d0-4409-afd0-10f4aba7aae5)

**Comment by RandomLonelyFox** [comment]
Weird, I've followed all  the steps mentioned on the main page, even made a new save file -with and without other mods- to check if it happens there (it does) and even updated to the newest version and yet i still get the same error message, just with the addition of grubs now being included in the whole mess of a text-wall. The issue certainly appears to be on my side. Ive reinstalled every mod and even SMAPI itself... the only other thing i could think of would be to reinstall stardew valley itself and see if that fixes the issue...
Additionally I am still not quite sure what the error is in the first place. Nothing seems to actually change, combat works fine, there is no lag, no invisible monsters or other bugs, just SMAPI freaking out.

**Comment by daleao** [comment]
Try out version 4.0.0, although I don't expect there to be any difference.

**Comment by emurphy42** [comment]
Just upgraded from 3.x to 4.0.1 and got a similar issue. Year 1 day 8, Broken Trident, no professions, no obvious glitches or lag (just happened to notice some red text after finishing the day and closing down).

```
[20:04:33 ERROR game] An error occurred in the base update loop: NullReferenceException: Object reference not set to an instance of an object.
   at StardewValley.Monsters.Monster.update_PatchedBy<DaLion.Overhaul.Modules.Combat__DaLion.Overhaul.Modules.Professions>(Monster this, GameTime time, GameLocation location)
   at StardewValley.Monsters.GreenSlime.update_PatchedBy<DaLion.Overhaul.Modules.Professions>(GreenSlime this, GameTime time, GameLocation location)
   at StardewValley.GameLocation.updateCharacters_PatchedBy<DaLion.Overhaul.Modules.Combat>(GameLocation this, GameTime time)
   at StardewValley.GameLocation.updateEvenIfFarmerIsntHere_PatchedBy<DaLion.Overhaul.Modules.Combat>(GameLocation this, GameTime time, Boolean ignoreWasUpdatedFlush)
   at StardewValley.Locations.MineShaft.updateEvenIfFarmerIsntHere(GameTime time, Boolean ignoreWasUpdatedFlush)
   at StardewValley.Locations.MineShaft.UpdateMines(GameTime time)
   at StardewValley.Game1.UpdateLocations(GameTime time)
   at StardewValley.Game1._update(GameTime gameTime)
   at StardewValley.Game1.Update(GameTime gameTime)
   at StardewModdingAPI.Framework.SCore.OnPlayerInstanceUpdating(SGame instance, GameTime gameTime, Action runUpdate)

```

**Comment by daleao** [comment]
Need confirmation whether issue persists in 4.1.0.

**Comment by Renjeth** [comment]
Not solved for me. I've been experiencing this on every screen with an enemy on it since I started playing again a few weeks ago. I think that was on version 3.1.9 or 3.1.10. It has persisted through mod versions after.

I tried copying the game folder and reducing the mod list to purely the ones needed for this mod, and I still get a wall of red text in the log on any screen with an enemy on it.
Here's my log file:
https://smapi.io/log/cd3a26745dbb42da90d15fd6d4cc0bfa 

I even started a new game and warped to the mines. Still wall of red text on any screen with enemies on it.
Here's the log file from that:
https://smapi.io/log/daa90eca42d24b0581d021abf34cbd39

**Comment by daleao** [comment]
Since I cannot replicate this on my setup, and this is clearly not a mod conflict or save-related issue, I need to ask you to please zip up *your entire game folder* (including the game itself, mods and smapi) and share it, so that I can test under the exact same circumstances.

**Comment by philredman** [comment]
I'm also getting this wall of text whenever slimes are around. Here's a log from the most recent 4.1.1 version:
https://smapi.io/log/f4b9bfaea80e4e34a240e9049dab67c1

Steps seem to be the same: 
1: enter map with slimes
2: wall of red text in SMAPI
3: no major negative effect, maybe a little bit of lag?

I've been getting this error for the entirety of using Modular Overhaul. I don't have the Combat module enabled, and I have limit breaks disabled. I've gotten it both before and after getting to level 10 combat, and before and after getting the slime profession.

I did install Modular Overhaul into an ongoing game, rather than starting a new game with Modular Overhaul enabled from the beginning. Maybe that's related?

**Comment by daleao** [comment]
Like I said above, at this point I need someone to share their entire game folder.

**Comment by RandomLonelyFox** [comment]
I've put my entire "Stardew Valley" folder into this zip, hope this helps?
https://www.dropbox.com/scl/fi/ahwg7ub8a6dromxbqdwch/Stardew-Valley.zip?rlkey=blo8dj43stewjrvo7d2kvru63&dl=0
It should be downloadable, but if its not ill see about uploading it somewhere else as dropbox has... issues sometimes.

**Comment by daleao** [comment]
It seems like having FTM installed fixes this issue.
I'm still not sure what the cause is.

**Comment by daleao** [comment]
Should finally be fixed in 4.1.2.
